gcloud/storage/bucket.py

```diff
@@ -854,18 +869,26 @@ def make_public(self, recursive=False, future=False):
         :type future: boolean
         :param future: If True, this will make all objects created in the
                        future public as well.
+
+        :type connection: :class:`gcloud.storage.connection.Connection` or
+                          ``NoneType``
+        :param connection: Optional. The connection to use when sending
+                           requests. If not provided, falls back to default.
         """
+        connection = _require_connection(connection)
+
         self.acl.all().grant_read()
-        self.acl.save()
+        self.acl.save(connection=connection)
 
         if future:
             doa = self.default_object_acl
             if not doa.loaded:
-                doa.reload()
+                doa.reload(connection=connection)
             doa.all().grant_read()
-            doa.save()
+            doa.save(connection=connection)
 
         if recursive:
-            for blob in self:
+            for blob in self.list_blobs(projection='full',
+                                        connection=connection):
                 blob.acl.all().grant_read()
-                blob.save_acl()
+                blob.acl.save(connection=connection)
```

Comment:
(Still somewhat outside the scope of this PR, but I'll keep going)

It's somewhat like `Bucket.delete(force=True)`. It works when it works, but can be a really bad idea if a user doesn't understand what they are doing.

- If we see this just as a method for buckets used as websites, we should rename
- We should strongly consider capping the number of blobs that can be iterated through, and maybe allow it to be overridden as an argument to the method